### ISSUE TITLE
[api-documenter] Configure `newLineKind` in api-documentor

### ISSUE THREAD

**Issue by matteematt** [microsoft/rushstack#4048]
## Summary

I have configured using `api-extractor.json` to produce `LF` line endings
But we get an output where the line endings are `CLRF`

> there is a change I am just configuring it wrong so let me know if thatis the case

Configure using `newLineKind` the line endings to be `LF`.
Instead we are genearing `CLRF` line endings

## Repro steps

We have the following config
```
{
	"$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
	"mainEntryPointFilePath": "<projectFolder>/dist/dts/index.d.ts",
	"newlineKind": "lf",
	"apiReport": {
		"enabled": true,
		"reportFolder": "<projectFolder>/docs",
		"reportFileName": "api-report.md"
	},
	"docModel": {
		"enabled": true,
		"apiJsonFilePath": "<projectFolder>/dist/<unscopedPackageName>.api.json"
	},
	"dtsRollup": {
		"enabled": true
	},
	"compiler": {
		"tsconfigFilePath": "<projectFolder>/tsconfig.build.json"
	},
	"messages": {
		"extractorMessageReporting": {
			"default": {
				"logLevel": "warning"
			},
			"ae-unresolved-link": {
				"logLevel": "error"
			},
			"ae-different-release-tags": {
				"logLevel": "none"
			}
		}
	}
}
```

We then run these two npm commands
```
    "api:extract": "api-extractor run",
    "api:document": "api-documenter markdown -i dist -o docs/api",
```

  **Expected result:** 

File with `LF` line endings

  **Actual result:** 

`cat -e <filename>`
```
<!-- Do not edit this file. It is automatically generated by API Documenter. -->^M$
^M$
[Home](./index.md)^M$
^M$
## API Reference^M$
^M$
## Packages^M$
^M$
|  Package | Description |^M$
|  --- | --- |^M$
|  [package](./index.md) |  |^M$
^M$
```

`^M$` is cat signifying that the file has `CLRF`.

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/api-documenter` version? | 7.21.6 | 
| Operating system? | Mac |
| Documentation target? | Markdown |
| Would you consider contributing a PR? | yes |
| TypeScript compiler version? | 4.5.5 |
| Node.js version (`node -v`)? | v16.17.0 |

**Comment by matteematt** [comment]
I have tried debugging it. I think the issue is that the markdown action documenter has the config hard coded as `undefined` for some reason.

https://github.com/microsoft/rushstack/blob/97b135591fa5ccf4be7c45897ab002712aa6ecdb/apps/api-documenter/src/cli/MarkdownAction.ts#L25

Whereas if you look at the `GenerateAction` documenter, that loads the config and passes it into the documenter

https://github.com/microsoft/rushstack/blob/97b135591fa5ccf4be7c45897ab002712aa6ecdb/apps/api-documenter/src/cli/GenerateAction.ts#L25

I added a `console.trace()` and when I run my `api-documenter` function it runs using the one using hardcoded undefined

```
Trace
    at MarkdownAction.onExecute (/Users/matt.walker/genesis/foundation-ui/_worktrees_git/mw/FUI-1069-strict-configuration-of-lf-line-endings/node_modules/@microsoft/api-documenter/lib/cli/MarkdownAction.js:20:11)
    at MarkdownAction._execute (/Users/matt.walker/genesis/foundation-ui/_worktrees_git/mw/FUI-1069-strict-configuration-of-lf-line-endings/node_modules/@rushstack/ts-command-line/lib/providers/CommandLineAction.js:60:21)
    at ApiDocumenterCommandLine.onExecute (/Users/matt.walker/genesis/foundation-ui/_worktrees_git/mw/FUI-1069-strict-configuration-of-lf-line-endings/node_modules/@rushstack/ts-command-line/lib/providers/CommandLineParser.js:211:39)
    at ApiDocumenterCommandLine.executeWithoutErrorHandling (/Users/matt.walker/genesis/foundation-ui/_worktrees_git/mw/FUI-1069-strict-configuration-of-lf-line-endings/node_modules/@rushstack/ts-command-line/lib/providers/CommandLineParser.js:169:25)
    at ApiDocumenterCommandLine.execute (/Users/matt.walker/genesis/foundation-ui/_worktrees_git/mw/FUI-1069-strict-configuration-of-lf-line-endings/node_modules/@rushstack/ts-command-line/lib/providers/CommandLineParser.js:101:24)
    at Object.<anonymous> (/Users/matt.walker/genesis/foundation-ui/_worktrees_git/mw/FUI-1069-strict-configuration-of-lf-line-endings/node_modules/@microsoft/api-documenter/lib/start.js:39:8)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
```

**Comment by SkaffenAmtiskaw** [comment]
We have this same issue - as a workaround we're running prettier immediately afterwards but it would be nice to not have the extra step.

**Comment by matteematt** [comment]
@SkaffenAmtiskaw the fix seems relatively simple, I have created a PR for it #4049 but awaiting to hear back from the team

**Comment by dmurvihill** [comment]
I'd like to reopen this issue. The fix, #4049, was reverted because the team wanted to instead graduate and document `api-documenter generate`. However, the latter steps never occurred.

**Comment by matteematt** [comment]
@dmurvihill can you also comment on that linked PR I made please? That might be more visible to them as I actually don't think they ever commented on this issue themselves
